AgendaUWP/AgendaUWP/ViewModels/ContactListPageViewModel.cs

```diff
@@ -60,19 +71,37 @@ private void InitilizeCommands()
 
         private void GoToDetailContact(Contact contact)
         {
-            _navigationService.Navigate(PageTokens.DetailContactPage, contact);
+            _navigationService.Navigate(PageTokens.ContactDetailPage, contact);
         }
 
         private ObservableCollection<ContactItemsGroup> GetData()
         {
-            var contacts = _contactService.GetContacts();
-            GroupedContactItems = contacts
-                                    .OrderBy(c => c.FullName)
-                                    .GroupBy(c => c.FullName.FirstOrDefault(), (key, list) => new ContactItemsGroup(key, list));
-                                        
+            var contacts = _contactService.GetAll();
+
+            GroupedContactItems = GetGroupedContactItemsFromList(contacts);
+
             return new ObservableCollection<ContactItemsGroup>(GroupedContactItems);
 
         }
+
+        private void Search(string query)
+        {
+            var contacts = _contactService.GetAll();
+
+            query = query.ToLowerInvariant();
+            contacts = contacts.Where(obj => obj.FullName.ToLowerInvariant().Contains(query)).ToList();
```

Comment:
Call the method "ToList()" isn't necessary. Could you please remove it?

Reply:
Ok! Done!